unity-client/Assets/Scripts/MainScripts/DCL/Controllers/MemoryManager/MemoryManager.cs

```diff
@@ -6,28 +7,42 @@
 
 namespace DCL
 {
-    public class MemoryManager
+    public class MemoryManager : IDisposable
     {
         private const uint MAX_USED_MEMORY = 1300 * 1024 * 1024;
         private const float TIME_FOR_NEW_MEMORY_CHECK = 1.0f;
 
         private List<object> idsToCleanup = new List<object>();
 
+        private Coroutine autoCleanupCoroutine;
+
         public void Initialize()
         {
-            CoroutineStarter.Start(AutoCleanup());
+            autoCleanupCoroutine = CoroutineStarter.Start(AutoCleanup());
+        }
+
+        public void Dispose()
+        {
+            CleanupPoolsIfNeeded(true, immediate: true);
+
+            if (autoCleanupCoroutine != null)
+                CoroutineStarter.Stop(autoCleanupCoroutine);
```

Comment:
We should nullify `autoCleanupCoroutine`. It's harmless the way it is since we are only stopping the coroutine when disposing, but I rather have it nulled just in case.